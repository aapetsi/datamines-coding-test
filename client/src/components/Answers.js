import React, { Component } from "react";

class Answers extends Component {
  state = {
    value: "1"
  };

  onChange = e => {
    this.setState({ value: e.target.value });
  };

  render() {
    const { answers, getResponseId } = this.props;
    const responseOptions = answers.map(response => (
      <option
        onClick={() => getResponseId(this.state.value)}
        value={response.id}
        key={response.id}
      >
        {response.option}
      </option>
    ));
    return (
      <div>
        <select onChange={this.onChange} name="" id="">
          {responseOptions}
        </select>
      </div>
    );
  }
}

export default Answers;
