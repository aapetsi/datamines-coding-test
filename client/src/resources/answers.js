const answers = [
  {
    responderId: 41,
    responses: [
      {
        questionId: 2,
        responseId: 2
      },
      {
        questionId: 3,
        responseId: 1
      }
    ]
  },
  {
    responderId: 25,
    responses: [
      {
        questionId: 2,
        responseId: 1
      },
      {
        questionId: 3,
        responseId: 1
      }
    ]
  }
];

export default answers;
