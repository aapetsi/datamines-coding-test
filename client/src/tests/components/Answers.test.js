import React from "react";
import { shallow } from "../../enzyme";
import Answers from "../../components/Answers";
import answers from "../fixtures/answers";

test("should render Answers with answers", () => {
  const wrapper = shallow(<Answers answers={answers} />);
  expect(wrapper).toMatchSnapshot();
});
