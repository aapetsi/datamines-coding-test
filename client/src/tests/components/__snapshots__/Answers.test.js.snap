// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Answers with answers 1`] = `ShallowWrapper {}`;
